vcd/resource_vcd_catalog_media.go

```diff
@@ -211,18 +291,72 @@ func createOrUpdateMediaItemMetadata(d *schema.ResourceData, meta interface{}) e
 			}
 		}
 		for _, k := range toBeRemovedMetadata {
-			err := mediaItem.DeleteMetadata(k)
+			err := media.DeleteMetadata(k)
 			if err != nil {
 				return fmt.Errorf("error deleting metadata: %s", err)
 			}
 		}
 		// Add new metadata
 		for k, v := range newMetadata {
-			_, err = mediaItem.AddMetadata(k, v.(string))
+			_, err = media.AddMetadata(k, v.(string))
 			if err != nil {
 				return fmt.Errorf("error adding metadata: %s", err)
 			}
 		}
 	}
 	return nil
 }
+
+// resourceVcdCatalogMediaImport is responsible for importing the resource.
+// The following steps happen as part of import
+// 1. The user supplies `terraform import _resource_name_ _the_id_string_` command
+// 2. `_the_id_string_` contains a dot formatted path to resource as in the example below
+// 3. The functions splits the dot-formatted path and tries to lookup the object
+// 4. If the lookup succeeds it sets the ID field for `_resource_name_` resource in statefile
+// (the resource must be already defined in .tf config otherwise `terraform import` will complain)
+// 5. `terraform refresh` is being implicitly launched. The Read method looks up all other fields
+// based on the known ID of object.
+//
+// Example resource name (_resource_name_): vcd_catalog_media.my-media
+// Example import path (_the_id_string_): org.catalog.my-media-name
+func resourceVcdCatalogMediaImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	resourceURI := strings.Split(d.Id(), ".")
+	if len(resourceURI) != 3 {
+		return nil, fmt.Errorf("resource name must be specified as org.catalog.my-media-name")
+	}
+	orgName, catalogName, mediaName := resourceURI[0], resourceURI[1], resourceURI[2]
+
+	if orgName == "" {
+		return nil, fmt.Errorf("import: empty org name provided")
+	}
+	if catalogName == "" {
+		return nil, fmt.Errorf("import: empty catalog name provided")
+	}
+	if mediaName == "" {
+		return nil, fmt.Errorf("import: empty media item name provided")
+	}
+
+	vcdClient := meta.(*VCDClient)
+	adminOrg, err := vcdClient.GetAdminOrgByName(orgName)
+	if err != nil {
+		return nil, fmt.Errorf(errorRetrievingOrg, orgName)
+	}
+
+	catalog, err := adminOrg.GetCatalogByName(catalogName, false)
+	if err != nil {
+		return nil, govcd.ErrorEntityNotFound
+	}
+
+	media, err := catalog.GetMediaByName(mediaName, false)
+	if err != nil {
+		return nil, govcd.ErrorEntityNotFound
+	}
+
+	_ = d.Set("org", orgName)
+	_ = d.Set("catalog", catalogName)
+	_ = d.Set("name", mediaName)
+	_ = d.Set("description", media.Media.Description)
+	d.SetId(media.Media.ID)
```

Comment:
See my comment in the `Read` function. We are not setting several attributes because we lack the data.

Reply:
Comment in above